tests/cli-misc.rs

```diff
@@ -445,7 +445,31 @@ fn rls_exists_in_toolchain() {
 }
 
 #[test]
-fn rls_does_not_exist_in_toolchain() {
+fn run_rls_when_not_available_in_toolchain() {
+    clitools::setup(Scenario::UnavailableRls, &|config| {
+        set_current_dist_date(config, "2015-01-01");
+        expect_ok(config, &["rustup", "default", "nightly"]);
+        expect_err(
+            config,
+            &["rls", "--version"],
+            &format!(
+                "the 'rls' component which provides the command 'rls{}' is not available for the 'nightly-{}' toolchain",
+                EXE_SUFFIX,
+                this_host_triple(),
+            ),
+        );
+
+        set_current_dist_date(config, "2015-01-02");
+        expect_ok(config, &["rustup", "update", "--no-self-update"]);
+        expect_ok(config, &["rustup", "component", "add", "rls"]);
+
+        assert!(config.exedir.join(format!("rls{}", EXE_SUFFIX)).exists());
```

Comment:
This assert checks for the proxy, which you already know exists because of the `expect_err()` call above succeeding in order to get this far.  As such either it should be moved to the top of this test, or just dropped.

Reply:
Ah, I was thinking that the proxies didn't exist until being installed. But that does clarify some things for sure.